### ISSUE TITLE
After installing gnome3, XDG_CURRENT_DESKTOP is still KDE.

### ISSUE THREAD

**Issue by Poetastrophe** [NixOS/nixpkgs#97523]
**Describe the bug**
After installing gnome3 declaratively in the configuration.nix file, XDG_CURRENT_DESKTOP is not changed from my previous KDE DE. 
Leading to unexpected behaviour for xdg-open (since it tries to use kde-open5).

**To Reproduce**
0. Have this in your configuration.nix:
```
services.xserver.enable = true;
services.xserver.desktopManager.plasma5.enable = true;
```
1. change it to and rebuild:
```
services.xserver.enable = true;
services.xserver.desktopManager.gnome3.enable = true;
services.xserver.displayManager.gdm.enable = true;
```
2. Type the following in your terminal
```
$ echo $XDG_CURRENT_DESKTOP
```

**Expected behavior**
The terminal prints something like "Gnome3"

** What happenes instead **
It prints "KDE".

**Screenshots**
![image](https://user-images.githubusercontent.com/17691932/92584615-25b63080-f294-11ea-92d0-1a6093e20023.png)
I am on gnome :(

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.42, NixOS, 20.03.1998.93b68c7cc5a (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.5`
 - channels(chris): `"home-manager"`
 - channels(root): `"nixos-20.03.1998.93b68c7cc5a"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by worldofpeace** [comment]
Yeah, this is incredibly confusing. I've never seen this happen before :thinking: My only thoughts is that something has to be setting `XDG_CURRENT_DESKTOP`, otherwise it wouldn't reach that state.

**Comment by worldofpeace** [comment]
cc @nixos/gnome

**Comment by jtojnar** [comment]
Does the variable still persist after reboot?

**Comment by Artturin** [comment]
this isn't nixos specific
